core/src/test/java/org/mobilitydata/gtfsvalidator/table/AnyTableLoaderTest.java

```diff
@@ -0,0 +1,216 @@
+package org.mobilitydata.gtfsvalidator.table;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.*;
+import static org.mobilitydata.gtfsvalidator.TestUtils.*;
+import static org.mockito.Mockito.*;
+
+import com.google.common.collect.ImmutableList;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import org.junit.Before;
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.annotation.FieldLevelEnum;
+import org.mobilitydata.gtfsvalidator.input.CountryCode;
+import org.mobilitydata.gtfsvalidator.input.CurrentDateTime;
+import org.mobilitydata.gtfsvalidator.notice.*;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsStop;
+import org.mobilitydata.gtfsvalidator.testgtfs.GtfsStopTableDescriptor;
+import org.mobilitydata.gtfsvalidator.validator.DefaultValidatorProvider;
+import org.mobilitydata.gtfsvalidator.validator.ValidationContext;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorLoader;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorProvider;
+
+public class AnyTableLoaderTest {
+
+  ValidationContext validationContext;
+
+  @Before
+  public void setup() {
+    validationContext =
+        ValidationContext.builder()
+            .setCountryCode(CountryCode.forStringOrUnknown("CA"))
+            .setCurrentDateTime(new CurrentDateTime(ZonedDateTime.now(ZoneId.systemDefault())))
+            .build();
+  }
+
+  @Test
+  public void invalidInputStream() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("_not_a_valid_file_");
+    NoticeContainer loaderNotices = new NoticeContainer();
+
+    AnyTableLoader.load(
+        testTableDescriptor,
+        mock(ValidatorProvider.class),
+        retrieveInputStream("_not_a_valid_file_"),
+        loaderNotices);
+
+    assertTrue(loaderNotices.hasValidationErrors());
+
+    assertValidationNotice(loaderNotices, CsvParsingFailedNotice.class);
+    assertEquals(1, loaderNotices.getValidationNotices().size());
```

Comment:
In general, I'm not a big fan of test assertion helpers where the assertion failure is in a test helper method.  I recognize why it's tricky to validation the notices directly here, but I'd propose a helper like the following:

```
// In TestUtils...
public static List<Class<? extends ValidationNotice>> validationNoticeTypes(NoticeContainer notices) {
  return notices.getValidationNotices().stream().map(Object::getClass).collect(toList());
}
```

And then the three test lines on 52, 54, and 55 could replaced with a single line:

```
assertThat(validationNoticeTypes(loaderNotices)).containsExactly(CsvParsingFailedNotice.class);
```

that would be more expressive in the failure case and trigger within the test itself.

I think

Reply:
Good suggestion!